peekingduck/pipeline/nodes/base.py

```diff
@@ -36,117 +37,75 @@ class ThresholdCheckerMixin:
     values, typically thresholds.
     """
 
-    def check_bounds(
-        self,
-        key: Union[str, List[str]],
-        value: Union[Number, Tuple[Number, Number]],
-        method: str,
-        include: Optional[str] = "both",
-    ) -> None:
-        """Checks if the configuration value(s) specified by `key` satisties
+    interval_pattern = re.compile(
+        r"^[\[\(]\s*[-+]?(inf|\d*\.?\d+)\s*,\s*[-+]?(inf|\d*\.?\d+)\s*[\]\)]$"
```

Comment:
I had some trouble deciding what the regex pattern should be responsible of filtering.
Based on [wiki](https://en.wikipedia.org/wiki/Interval_(mathematics)#Including_or_excluding_endpoints) `(a, a)` is a valid (empty) interval, therefore, `(+inf, +inf)` or `[+inf, +inf]` is a valid format.

There is an `if` statement below to catch cases where the lower bound is larger than the upper bound.

Reply:
Ah, good point about `(a, a)` where `a` can be `-inf` / `+inf`.
Since there's the `if` statement to detect invalid bounds, can leave the regex as is.